test-app-reactive/src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagenameasync/service/MessageListener.kt

```diff
@@ -2,16 +2,21 @@ package uk.gov.justice.digital.hmpps.hmppstemplatepackagenameasync.service
 
 import com.fasterxml.jackson.databind.ObjectMapper
 import io.awspring.cloud.sqs.annotation.SqsListener
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.future.asCompletableFuture
+import kotlinx.coroutines.launch
 import org.springframework.stereotype.Service
+import java.util.concurrent.CompletableFuture
 
 @Service
 class InboundMessageListener(private val inboundMessageService: InboundMessageService, private val objectMapper: ObjectMapper) {
 
   @SqsListener(queueNames = ["inboundqueue"], factory = "hmppsQueueContainerFactoryProxy")
-  fun processMessage(rawMessage: String?) {
+  fun processMessage(rawMessage: String?): CompletableFuture<Unit> {
```

Comment:
making this return a `CompletableFuture` then means we get an async message listener instead - https://docs.awspring.io/spring-cloud-aws/docs/3.0.0-M3/reference/html/index.html#receiving-messages